newIDE/app/src/GameDashboard/GameAnalyticsEvaluator.js

```diff
@@ -0,0 +1,653 @@
+      metric.sessions && metric.sessions.d0SessionsDurationTotal !== null
+        ? metric.sessions.d0SessionsDurationTotal
+        : undefined;
+
+    const d0Players = metric.players ? metric.players.d0Players : 0;
+    const d0PlayersBelow60s =
+      metric.players && metric.players.d0PlayersBelow60s !== null
+        ? metric.players.d0PlayersBelow60s
+        : undefined;
+    const d0PlayersBelow180s =
+      metric.players && metric.players.d0PlayersBelow180s !== null
+        ? metric.players.d0PlayersBelow180s
+        : undefined;
+    const d0PlayersBelow300s =
+      metric.players && metric.players.d0PlayersBelow300s !== null
+        ? metric.players.d0PlayersBelow300s
+        : undefined;
+    const d0PlayersBelow600s =
+      metric.players && metric.players.d0PlayersBelow600s !== null
+        ? metric.players.d0PlayersBelow600s
+        : undefined;
+    const d0PlayersBelow900s =
+      metric.players && metric.players.d0PlayersBelow900s !== null
+        ? metric.players.d0PlayersBelow900s
+        : undefined;
+
+    playersBelowSums[durationIndexes.for1Minute] += d0PlayersBelow60s || 0;
+    playersBelowSums[durationIndexes.for3Minutes] += d0PlayersBelow180s || 0;
+    playersBelowSums[durationIndexes.for5Minutes] += d0PlayersBelow300s || 0;
+    playersBelowSums[durationIndexes.for10Minutes] += d0PlayersBelow600s || 0;
+    playersBelowSums[durationIndexes.for15Minutes] += d0PlayersBelow900s || 0;
+    playedDurationSumInMinutes += d0SessionsDurationTotal || 0;
+    playersSum += d0Players;
+    onlyFullyDefinedPlayersSum +=
+      d0PlayersBelow60s !== undefined ? d0Players : 0;
+  });
+  playedDurationSumInMinutes /= 60;
+
+  const viewersCount = onlyFullyDefinedPlayersSum;
+  const playersCount =
+    onlyFullyDefinedPlayersSum - playersBelowSums[durationIndexes.for1Minute];
+
+  const nearestToMedianDurationIndex = findNearestToMedianDurationIndex(
+    playersBelowSums,
+    playersCount
+  );
+  const greaterDurationPlayerIndex = findGreaterDurationPlayerIndex(
+    playersBelowSums,
+    viewersCount
+  );
+
+  const dateFormatOptions = { month: 'short', day: 'numeric' };
+  const noMonthDateFormatOptions = { day: 'numeric' };
+
+  const formatDate = (metric: MergedGameMetrics) => {
+    const startIsoDate = metric.startDate;
+    const endDate = parseISO(metric.date);
+    const formattedDate = endDate.toLocaleDateString(
+      undefined,
+      dateFormatOptions
+    );
+    if (!startIsoDate) {
+      return formattedDate;
+    }
+    const startDate = parseISO(startIsoDate);
+    return (
+      startDate.toLocaleDateString(
+        undefined,
+        endDate.getMonth() === startDate.getMonth()
+          ? noMonthDateFormatOptions
+          : dateFormatOptions
+      ) +
+      '-' +
+      formattedDate
+    );
+  };
+
+  return {
+    overview: {
+      // Players from before the migration are shown as viewers
+      // because it's not possible to make them apart from real players.
```

Comment:
should we add a TODO here to change this once we have the data? with a date maybe too?

Reply:
The date is in a comment on the back-end side.
There will be many things to change and I'm not sure if we'll do it one day so I prefer not not add TODO comments to avoid confusion.